opentelemetry-sdk/src/opentelemetry/sdk/metrics/export/aggregate.py

```diff
@@ -30,24 +31,43 @@ class Aggregator(abc.ABC):
     """
 
     def __init__(self, config=None):
-        self.current = None
-        self.checkpoint = None
-        if config:
+        self._lock = threading.Lock()
+        self.last_update_timestamp = 0
+        self.last_checkpoint_timestamp = 0
+        if config is not None:
             self.config = config
         else:
             self.config = {}
 
     @abc.abstractmethod
     def update(self, value):
         """Updates the current with the new value."""
+        self.last_update_timestamp = time_ns()
 
     @abc.abstractmethod
     def take_checkpoint(self):
         """Stores a snapshot of the current value."""
+        self.last_checkpoint_timestamp = time_ns()
 
     @abc.abstractmethod
     def merge(self, other):
         """Combines two aggregator values."""
+        self.last_update_timestamp = max(
```

Comment:
Can we safely assume that all aggregators want this behaviour as the default? `LastValueAggregator` already doesn't follow this.

Reply:
Actually, this is not the default behavior, it is just common code that all (or most) aggregators can use if they decide so, by calling `super` themselves.